feat(cli): add commit hash and target to long_version output

kt3k opened the pull request:
This PR adds the git commit hash and build target to long version output (`--version`) as suggested in #6037. The output now looks like the below:

```shellsession
$ ./target/debug/deno --version
deno 1.4.6 (0dcdc5e)
target x86_64-apple-darwin
v8 8.7.220.3
typescript 4.0.3
```

---
fixes #6037

ry commented on cli/version.rs:
```diff
@@ -1,6 +1,7 @@
 // Copyright 2018-2020 the Deno authors. All rights reserved. MIT license.
 
 pub const DENO: &str = env!("CARGO_PKG_VERSION");
+pub const BUILD: &str = env!("GIT_COMMIT_HASH");
```
Just "GIT_COMMIT_HASH" instead of "BUILD" would be more descriptive.

ry commented on cli/flags.rs:
```diff
@@ -221,8 +221,10 @@ To evaluate code in the shell:
+    "{} ({})\ntarget {}\nv8 {}\ntypescript {}",
     crate::version::DENO,
+    crate::version::BUILD,
+    env!("TARGET"),
```
It's a little odd that the target is on its own line. Maybe it should be in the parens too?

kt3k replied:
How about this format?

```shellsession
$ deno --version
deno 1.4.6 (0dcdc5e, x86_64-apple-darwin)
v8 8.7.220.3
typescript 4.0.3
```

ry replied:
I think that makes more sense. It would be nice to also know if we're using release/debug build.

lucacasonato commented on cli/build.rs:
```diff
@@ -104,6 +104,18 @@ fn ts_version() -> String {
     .collect::<String>()
 }
 
+fn git_commit_hash() -> String {
+  let output = std::process::Command::new("git")
+    .arg("rev-list")
+    .arg("-1")
+    .arg("HEAD")
+    .output()
+    .expect("failed to execute process");
+  std::str::from_utf8(&output.stdout[..7])
+    .unwrap()
+    .to_string()
+}
+
```
Is this going to work when doing `cargo install deno`?

ry replied:
It's a good point. I think it will break "cargo install deno".

kt3k replied:
I see. It won't work with `cargo install`...

We need a different strategy here.

kt3k replied:
It seems sha1 is stored in the file called `.cargo_vcs_info.json` in crates.io. (ref: https://github.com/rust-lang/cargo/issues/5629 ) Maybe we can use it.

lucacasonato replied:
Maybe just return `"crates.io"` if anything in `git_commit_hash` fails?

kt3k commented:
Closing for the moment. I'll reopen it when I find a way to handle `cargo install deno` case correctly.

bartlomieju commented:
@kt3k I think this is a valid approach. As @lucacasonato suggested we can fallback to some constant value eg. empty string. For reference [`vergen`](https://github.com/rustyhorde/vergen) returns `UNKNOWN` if `git` command fails (https://github.com/rustyhorde/vergen/blob/8d522db8c8e16e26c0fc9ea8e6b0247cbf5cca84/src/output/mod.rs#L88-L95).

kt3k commented:
OK. I'll update the change!

kt3k commented:
I updated the change. The commit hash becomes `UNKNOWN` when not in git repository. The output looks like the below (I also added `PROFILE`):

```shellsession
$ ./target/debug/deno --version
deno 1.5.0 (UNKNOWN, debug, x86_64-apple-darwin)
v8 8.7.220.3
typescript 4.0.3
```

bartlomieju commented:
The solution looks satisfactory to me. Waiting on @ry to double check before landing.

ry reviewed (approved):
LGTM - thank you @kt3k !
